addons/addon-base-raas-ui/packages/base-raas-ui/src/models/environments-sc/ScEnvironmentsStore.js

```diff
@@ -130,6 +157,22 @@ const ScEnvironmentsStore = BaseStore.named('ScEnvironmentsStore')
       return self.environments.get(id);
     },
 
+    canChangeState(id) {
+      const outputs = self.environments.get(id).outputs;
+      let result = false;
+      // TODO: Remove isRStudio check after CNAME patch
+      let isRStudio = false;
+      outputs.forEach(output => {
+        if (output.OutputKey === 'Ec2WorkspaceInstanceId' || output.OutputKey === 'NotebookInstanceName') {
+          result = true;
+        }
+        if (output.OutputKey === 'MetaConnection1Type') {
+          isRStudio = true;
```

Comment:
It's not necessary that every workspace with a MetaConnection1Type output is an RStudio. 

Here you can only determine if this is an RStudio environment if an output with Key 'MetaConnection1Type' exists AND its Value is 'RStudio'